scripts/live_test/generate_index.py

```diff
@@ -0,0 +1,144 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+"""
+Generate index.html of testing results HTML pages.
+"""
+import traceback
+import os
+import requests
+import xml.etree.ElementTree as ET
+
+
+def generate(container, container_url, testdata, USER_REPO, USER_BRANCH, COMMIT_ID, USER_LIVE):
+    """
+    Generate index.html. Upload it to storage account
+    :param container:
+    :param container_url:
+    :return:
+    """
+    data = []
+    url = container_url + '?restype=container&comp=list'
+    content = requests.get(url).content
+    # print(content)
+    root = ET.fromstring(content)
+    for blobs in root:
+        for blob in blobs:
+            name = url = ''
+            for e in blob:
+                if e.tag == 'Name':
+                    name = e.text
+                if e.tag == 'Url':
+                    url = e.text
+            if name == '' or url == '':
+                print('Warning, name: {}, url: {}'.format(name, url))
+            if name.endswith('.html'):
+                data.append({'name': name, 'url': url})
+        break
+    print(data)
+    html = render(data, container_url, testdata, USER_REPO, USER_BRANCH, COMMIT_ID, USER_LIVE)
+    with open('index.html', 'w') as f:
+        f.write(html)
+
+    # Upload to storage account
+    cmd = 'az storage blob upload -f index.html -c {} -n index.html --account-name clitestresultstac'.format(container)
+    print('Running: ' + cmd)
+    os.system(cmd)
+
+
+def render(data, container_url, testdata, USER_REPO, USER_BRANCH, COMMIT_ID, USER_LIVE):
+    content = """
```

Comment:
how about extract it to a seperate file?

Reply:
144 lines. Short file. Don't worry.

Reply:
https://jinja.palletsprojects.com/en/2.11.x/ is a good choice. But it costs more time to write code. The current content is simple. If it gets complex enough, we can use template engine.